integration/ruler_test.go

```diff
@@ -293,6 +293,7 @@ func TestRulerEvaluationDelay(t *testing.T) {
 	require.NoError(t, mimir.WaitSumMetrics(e2e.Greater(ruleEvaluationsAfterPush[0]+float64(samplesToSend)), "cortex_prometheus_rule_evaluations_total"))
 
 	// query all results to verify rules have been evaluated correctly
+	t.Log("querying from ", now.Add(-evaluationDelay), "to", now)
```

Comment:
Another approach to this would be to add another assertion on the metric `cortex_prometheus_last_evaluation_samples` instead of `cortex_prometheus_rule_evaluations_total`. 

[This metric is incremented](https://github.com/prometheus/prometheus/blob/main/rules/group.go#L449) before the [samples are pushed](https://github.com/prometheus/prometheus/blob/main/rules/group.go#L482).

Reply:
I suspect that the problem is that we ignore the sample even though it was pushed.

Pushed:

```
2, 4, 6, 8, 10, 12, 14, 16, 18, 20
```

Queried:

```
4, 6, 8, 10, 12, 14, 16, 18, 20
```

The failure happens because the first sample we query is `4` and not `2`. TO me that fact that `4` exists means that `2` must exist as well, we just don't get it in the query results.